cache_flat_mutation_reader.hh

```diff
@@ -515,55 +483,102 @@ bool cache_flat_mutation_reader::ensure_population_lower_bound() {
-                _last_row->set_continuous(true);
+                if (_current_tombstone != _last_row->range_tombstone() && !_last_row->dummy()) {
+                    with_allocator(_snp->region().allocator(), [&] {
+                        auto e2 = alloc_strategy_unique_ptr<rows_entry>(
+                                current_allocator().construct<rows_entry>(table_s,
+                                                                          position_in_partition_view::before_key(_last_row->position()),
+                                                                          is_dummy::yes,
+                                                                          is_continuous::yes));
+                        // Use _next_row iterator only as a hint, because there could be insertions after _upper_bound.
+                        auto insert_result = rows.insert(std::move(e2), table_cmp);
+                        if (insert_result.second) {
+                            clogger.trace("csm {}: L{}: inserted dummy at {}", fmt::ptr(this), __LINE__, insert_result.first->position());
+                            _snp->tracker()->insert(*insert_result.first);
+                        }
+                        clogger.trace("csm {}: set_continuous({}), prev={}, rt={}", fmt::ptr(this), insert_result.first->position(),
+                                      _last_row.position(), _current_tombstone);
+                        insert_result.first->set_continuous(true);
+                        insert_result.first->set_range_tombstone(_current_tombstone);
+                        clogger.trace("csm {}: set_continuous({})", fmt::ptr(this), _last_row.position());
+                        _last_row->set_continuous(true);
+                    });
+                } else {
+                    clogger.trace("csm {}: set_continuous({}), rt={}", fmt::ptr(this), _last_row.position(), _current_tombstone);
+                    _last_row->set_continuous(true);
+                    _last_row->set_range_tombstone(_current_tombstone);
+                }
             } else {
-                clogger.trace("csm {}: set_continuous({})", fmt::ptr(this), e.position());
-                e.set_continuous(true);
+                if (_current_tombstone != e.range_tombstone() && !e.dummy()) {
+                    with_allocator(_snp->region().allocator(), [&] {
+                        auto e2 = alloc_strategy_unique_ptr<rows_entry>(
+                                current_allocator().construct<rows_entry>(table_s,
+                                                                          position_in_partition_view::before_key(e.position()),
+                                                                          is_dummy::yes,
+                                                                          is_continuous::yes));
+                        // Use _next_row iterator only as a hint, because there could be insertions after _upper_bound.
```

Comment:
This comment here is actually also copy-pasta and not applicable, since maybe_update_continuity() is called when _next_row is in range, to _upper_bound is after _next_row.

It's still a hint, since there could be insertions before _next_row.get_iterator_in_latest_version(), by concurrent reads, or earlier in this method by _next_row.ensure_entry_in_latest()

I'll update it to:

```
                        // Use _next_row iterator only as a hint because there could be insertions before 
                        // _next_row.get_iterator_in_latest_version(), either from concurrent reads,
                        // from _next_row.ensure_entry_in_latest().
```